src/events.js

```diff
@@ -0,0 +1,67 @@
+import { objectType, inArray } from "./core/utilities";
+
+const LISTENERS = Object.create( null );
+const SUPPORTED_EVENTS = [
+	"runStart",
+	"suiteStart",
+	"testStart",
+	"assertion",
+	"testEnd",
+	"suiteEnd",
+	"runEnd"
+];
+
+/**
+ * Emits an event with the specified data to all currently registered listeners.
+ * Callbacks will fire in the order in which they are registered (FIFO). This
+ * function is not exposed publicly; it is used by QUnit internals to emit
+ * logging events.
+ *
+ * @private
+ * @method emit
+ * @param {String} eventName
+ * @param {Object} data
+ * @return {Void}
+ */
+export function emit( eventName, data ) {
+	if ( objectType( eventName ) !== "string" ) {
+		throw new TypeError( "eventName must be a string when emitting an event" );
+	}
+
+	// Clone the callbacks in case one of them registers a new callback
+	let originalCallbacks = LISTENERS[ eventName ];
+	let callbacks = originalCallbacks ? [ ...originalCallbacks ] : [];
```

Comment:
It doesn't improve readability, but I was playing with the syntax: `[ ...(originalCallbacks || []) ]`. 

I still prefer your code. 👍